src/utils/graphql.js

```diff
@@ -1,35 +1,52 @@
-// import { createProvider } from './vue-apollo'
-import ApolloClient from 'apollo-boost'
-import { ApolloLink, concat } from 'apollo-link'
-import axios from 'axios'
-import { buildAxiosFetch } from 'axios-fetch'
+import ApolloClient from 'apollo-client'
+import { ApolloLink, split } from 'apollo-link'
+import { HttpLink } from 'apollo-link-http'
+import { WebSocketLink } from 'apollo-link-ws'
+import { getMainDefinition } from 'apollo-utilities'
+import { InMemoryCache } from 'apollo-cache-inmemory'
 
-export function createApolloClient (uri) {
-  const authMiddleware = new ApolloLink((operation, forward) => {
-    operation.setContext({
-      headers: {
-        // FIXME: this is the random generated password, update every time it is generated for now!
-        Authorization: 'Basic Y3lsYzpiaFRTbnRpWjRqR2YzSWR3eTlVMg=='
-      }
-    })
-    return forward(operation)
+/**
+ * Create an ApolloClient using the given URI's.
+ *
+ * If a `queryUri` is provided, it will be used for handling Query operations.
+ *
+ * If a `subscriptionUri` is provided, it will be used for handling Subscription operations.
+ *
+ * If no `subscriptionUri` is provided, any Subscription operation will fail, as we will be
+ * using an empty link (a simple instance of `ApolloLink`).
+ *
+ * The link object is actually a split function (from the `apollo-link` module). This function
+ * works similarly to a ternary operator. Based on the operation, it will return a Query or
+ * a Subscription link.
+ *
+ * @param queryUri {string} Query URI, e.g. http://localhost:3000/graphql
+ * @param subscriptionUri {object} Subscription URI, e.g. ws://localhost:3000/subscriptions
+ * @returns {DefaultClient} an ApolloClient
+ */
+export function createApolloClient (queryUri, subscriptionUri = null) {
+  const httpLink = new HttpLink({
+    uri: queryUri
   })
-  const client = new ApolloClient({
-    // You should use an absolute URL here
-    uri: uri,
-    // we must define fetch, as otherwise running unit tests it would fail as there is no global fetch variable
-    fetch: buildAxiosFetch(axios)
-  })
-  client.link = concat(authMiddleware, client.link)
-  return client
-}
 
-// GraphQL
-const apolloClient = new ApolloClient({
-  // You should use an absolute URL here
-  uri: 'https://api.graphcms.com/simple/v1/awesomeTalksClone',
-  // we must define fetch, as otherwise running unit tests it would fail as there is no global fetch variable
-  fetch: buildAxiosFetch(axios)
-})
+  const wsLink = subscriptionUri !== null ? new WebSocketLink({
+    uri: subscriptionUri,
+    options: {
+      reconnect: true
+    }
+  }) : new ApolloLink() // return an empty link, useful for testing
+
+  const link = split(
```

Comment:
Think of `split` as a ternary `?` operator. Given a condition, return either the next value, or the subsequent.